misago/legal/models.py

```diff
@@ -1 +1,114 @@
-# Empty models file that triggers migrations for this app
+from django.db import models
+from django.utils import timezone
+from django.core.exceptions import ValidationError
+from django.template.defaultfilters import slugify
+from django.utils.translation import ugettext_lazy as _
+
+from misago.conf import settings
+
+
+class NoActiveLegalAgreement(ValidationError):
+    pass
+
+
+class LegalAgreementManager(models.Manager):
+    def get_current(self):
+        try:
+            return self.get(is_active=True)
```

Comment:
What if there is more than one active agreement?

Reply:
There can be just one active type of agreement (terms_of_service or  privacy_policy). This will be ensured by 
```
if  self.is_active:
  LegalAgreement.objects.exclude(id=self.id).filter(title=self.title).update(is_active=False)
```
But I don't know whether you want to handle the case, when somebody wants to create multiple versions of the same legal type on the same date.This wont work at the moment, because I sticked  `key` to `create_at` and made it `unique`

Reply:
Have you read my post from 14 days ago? In it I have proposed that there may be any number of active agreements at single time.

Reply:
sorry @rafalp .

Reply:
@burton1982 there is no need to apologize for anything ;)

Reply:
I'll post acceptance criteria for this feature after I finish my work on IP stores.

Reply:
The Situation is these at the moment:
You can have infinitely active agreements on the same time as long the differ from the type.  I am just wondering why you would have two active `terms-of-service` or `privacy-policy` on the same time? 

As I am reading you comment from 14 days ago , I believe this here is just a misunderstanding

Reply:
This is correct, there shouldn't be situation when there is more than one active `terms-of-service` or `privacy-policy` agreement, but those two are not exclusive. I will have active `terms-of-service` and `privacy-policy` at single time.

Reply:
Cool, because this what this patch does. You can have active `terms-of-service` and `privacy-policy`  on the same time. In my patch the logic will set `is_active` to False to all of the same agreement type except this one you have marked as `active`. The other agreement types will keep untouched.

As I said: The only problem is that you can `release` only one version of a `agreement type` per day. In my opinion this should be okay, but the exception has to be catch up by the fronted.

Reply:
> and your code with throw MutlipleObjectsReturnedError when this is the case ;)
This is how I use it:
```
    def has_user_agreed_latest_legal(self, user, legal):
        return UserAgreementAcceptance.objects.filter(
            terms_of_service=LegalAgreement.objects.filter(title=legal).get_current(),
            user=user,
        ).exists()
```
Let me think a little about it.

Reply:
`LegalAgreement.objects.filter(title=legal).get_current()`

Can you confirm that this queryset actually works? You are defining `get_current()` on `Manager` and `objects.filter()` returns `Queryset` that shouldn't have this method.

Reply:
Also, remember that your code has to scale and cache from day one. Ideally, there should be utility module `misago.legal.utils` with methods like `get_active_agreements()` that should do one db hit or cache hit at most, and `get_missing_agreements(request)` utility that would return list of agreements that we require but *user* didn't give.

Reply:
You are absolutely right. I fixed the `query`  - i had mixed up `get` and `filter` . Now I will get rid of the Manager and as you suggested put it in the `utility` module.